### ISSUE TITLE
On this page component: Links are not setting focus correctly with iOS Safari + VoiceOver

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/vets-design-system-documentation#1979]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

While retesting an older issue, I noticed that on mobile using Safari and VoiceOver, the "On this page" links do not always set focus properly on the deep-linked headings. The first link is working as expected, but the links after were not, the focus was going to an element that was not the appropriate heading.

## What I expected to happen

I would expect that for any link I activate in the On this page component, the focus would be directed to the appropriate heading and the screen reader would announce the text of that heading.

## Reproducing
- Formation version:
- Device: iPhone 13
- Browser: Safari
- Screen reader: VoiceOver

Steps to reproduce:

1.Go to a a page with the component, for example VAMC health services page in Safari on iPhone (Example page: https://www.va.gov/eastern-oklahoma-health-care/health-services/)
2. Turn on iOS VoiceOver
3. Swipe right to the "On This Page" table of contents section
4. Continue swiping right to go to any of the links EXCEPT the first "Primary Care" link
5. Double-tap to activate one of these links and jump to the corresponding section of the page farther down
6. Notice that the focus goes to the last accordion in the previous section instead of the H2 heading that corresponds to the link you chose above. Your focus should have gone directly to the H2.

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [x] This bug is not blocking any work
- [ ] Other

**Comment by caw310** [comment]
@rsmithadhoc would you test to see if this is still an issue?

**Comment by amyleadem** [comment]
@caw310 
Confirming that this is still an issue.

**Details:** 
In VoiceOver Gestures, tapping on an anchor link inside the "On this page" component does move focus move to the accordion that directly precedes the expected header, rather than the header itself. For example, tapping on "Mental health care" should move focus to the "Mental health care" header. However, focus instead is moved to the "Women Veteran care" accordion button directly above it.
